File: TypeScript/Slice/add-element-to-slice.ts
function main(): void {
    // Create a slice.
    let randomData: string[] = ["a", "b", "c"]
    // Add an element to the slice, print the slice.
    console.log(addElementToSlice(randomData, "d"))
}

main()

// Add an element to the slice and return the new slice.
function addElementToSlice(originalSlice: string[], element: string): string[] {
    return originalSlice.concat(element)
}

File: Dart/Slice/add_element_to_slice.dart
import "dart:core";

void main() {
  // Create a new list of strings.
  final List<String> randomSlice = <String>["Bob", "Alice", "Joe"];
  // Add a new element to the slice.
  print(addElementToSlice(randomSlice, "Bill"));
}

// Add an element to the slice and return the slice.
List<dynamic> addElementToSlice(
  final List<dynamic> slice,
  final dynamic element,
) {
  slice.add(element);
  return slice;
}
